ctms/acoustic_service.py

```diff
@@ -0,0 +1,273 @@
+                    in AcousticResources.VALID_ACOUSTIC_MAIN_TABLE_FIELDS
+                ):
+                    acoustic_main_table[
+                        acoustic_field_name
+                    ] = self._transform_field_for_acoustic(value)
+                else:
+                    print(
+                        "Skipping CTMS field ctms_data[%s][%s] because no match in Acoustic"
+                        % (subdict, key)
+                    )
+
+        for key, value in ctms_data["email"].items():
+            acoustic_field_name = key
+            if acoustic_field_name == "primary_email":
+                acoustic_field_name = "email"
+            if (
+                acoustic_field_name
+                in AcousticResources.VALID_ACOUSTIC_MAIN_TABLE_FIELDS
+            ):
+                acoustic_main_table[
+                    acoustic_field_name
+                ] = self._transform_field_for_acoustic(value)
+            else:
+                print(
+                    "Skipping CTMS field ctms_data[email][%s] because no match in Acoustic"
+                    % (key)
+                )
+
+        # create the RT rows for the newsletter table in acoustic
+        newsletter_rows = []
+        for newsletter in ctms_data["newsletters"]:
+            newsletter_rows.append(
+                {
+                    "email_id": ctms_data["email"]["email_id"],
+                    "newsletter_name": newsletter["name"],
+                    "newsletter_format": newsletter["format"],
+                    "newsletter_lang": newsletter["lang"],
+                    "newsletter_source": newsletter["source"],
+                    "newsletter_unsub_reason": newsletter["unsub_reason"],
+                    "create_date": self._transform_field_for_acoustic(
+                        newsletter["create_timestamp"]
+                    ),
+                    "update_date": self._transform_field_for_acoustic(
+                        newsletter["update_timestamp"]
+                    ),
+                }
+            )
+            # and finally flip the main table's sub_<newsletter> flag to true for each subscription
+            if newsletter["subscribed"]:
+                acoustic_main_table[
+                    AcousticResources.MAIN_TABLE_SUBSCR_FLAGS[newsletter["name"]]
+                ] = "1"
+
+        return acoustic_main_table, newsletter_rows
+
+    @staticmethod
+    def _transform_field_for_acoustic(data):
+        if isinstance(data, bool):
+            if data:
+                return "1"
+            return "0"
+        # Acoustic doesn't have timestamps, so make timestamps into dates.
+        # Is this appropriate for all TS fields?
+        # Is the below RE appropriate for what we'll actually be getting from Postgres?
+        try:
+            matching = re.search(r"^(\d{4}-\d{2}-\d{2})T\d", data)
+            if matching:
+                return matching.groups()[0]
+        except TypeError:
+            pass
```

Comment:
I'm a bit confused by this. We're operating on data we got out of postgres, right? So this should be able to be operated on as whatever sqlalchemy gives us for a datetime? We can transform it to whatever string acoustic wants without the use of regex (unless I am not understanding what's going on here)

Reply:
sounds true, yeah.

Reply:
Attempted to address this in new commits